indico/modules/events/schemas.py

```diff
@@ -5,12 +5,54 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from marshmallow import fields
+import re
+
+from marshmallow import ValidationError, fields, validates_schema
 
 from indico.core.marshmallow import mm
 from indico.modules.events.models.events import Event
 from indico.modules.events.models.principals import EventPrincipal
-from indico.util.marshmallow import PrincipalPermissionList
+from indico.util.i18n import _
+from indico.util.marshmallow import PrincipalPermissionList, not_empty
+
+
+class ValidRegExField(fields.String):
+    """A valid Python regular expression."""
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if len(value) < 2:
+            raise ValidationError(_('Empty regex'))
+        try:
+            return re.compile(value).pattern
+        except ValueError as exc:
+            raise ValidationError(str(exc))
```

Comment:
```suggestion
        except (ValueError, re.error) as exc:
            raise ValidationError(_('Invalid regex: {}').format(exc))
```

not sure if `ValueError` is also needed, but `re.error` is the interesting one when you enter garbage

Reply:
Indeed. I think `ValueError` may have beenr relevant in some other incarnation of this code. Probably just forgot to change it.